### ISSUE TITLE
USB protection diodes

### ISSUE THREAD

**Issue by charlespax** [PaxInstruments/labwiz-board#76]
We are using some diodes on the USB lines. Make sure we are using the right diodes for this. Make sure we are using the right footprint.

**Comment by robojay** [comment]
Take a look at this app note from ST.  It covers ESD protection and not needing the series resistors you have in the current design.

[ST AN4879.pdf](https://github.com/PaxInstruments/labwiz-board/files/560836/ST.AN4879.pdf)

**Comment by charlespax** [comment]
Nice. We should ditch the resistors then. I also want to look at high-speed devices because the PCB will be compatible with several STM32Fx families.

![screen shot 2016-11-10 at 15 04 44](https://cloud.githubusercontent.com/assets/281642/20167525/181eeff2-a757-11e6-8b49-6d7116297031.png)

![screen shot 2017-01-27 at 15 48 29](https://cloud.githubusercontent.com/assets/281642/22363835/17135ace-e4a8-11e6-8c94-ba116b529988.png)

**Comment by charlespax** [comment]
Series resistors removed.
![screen shot 2017-01-27 at 14 57 01](https://cloud.githubusercontent.com/assets/281642/22362858/e608d596-e4a0-11e6-8aa2-b0c6d09a8a93.png)
![screen shot 2017-01-27 at 14 56 25](https://cloud.githubusercontent.com/assets/281642/22362843/cd691104-e4a0-11e6-8c2a-d1f8a7f0f784.png)

**Comment by charlespax** [comment]
- [x] "The full speed USB driver impedance on STM32 MCUs is guaranteed by design to be between 28 ohms and 44 ohms. The result is that these products do not need an external resistor on the board for the output driver impedance." Done, removed series resistors.
- [ ] ~~"The track routing should respect 90 ohm +/-15% differential from the STM32 MCU’s package to the receptacle. It is important to route the two data lines in parallel to ensure the best impedance matching. A 45 ohm +/-15% single ended trace is recommended."~~ New issue: #136
- [ ] ~~"Routing USB signals using a minimum of vias and corners reduces the signal reflections and the impedance changes. When it becomes necessary to turn by 90°, it is preferred to use two 45° turns or an arc instead of making a single 90° turn. This action reduces the reflections on the signal by minimizing the impedance discontinuities." We can do a lot better here.~~ New issue: #136
![screen shot 2017-01-27 at 15 50 01](https://cloud.githubusercontent.com/assets/281642/22363865/50a9ae28-e4a8-11e6-8ee5-df98e0d3839d.png)
- [x] "A protection against ESD and EMI is needed... The HBM requires that the USB pins of the component device are tolerant to up to 2 kV discharge, this is the case for the STM32 MCUs." Done. The ESD diodes can handle the 2 kv.
![screen shot 2017-01-27 at 15 45 27](https://cloud.githubusercontent.com/assets/281642/22363768/a9471774-e4a7-11e6-858c-f8dce4771185.png)
- [x] "The component needs to be placed as close as possible to the receptacle to add ESD protection against high ESD surge. The USBLC6 component is recommended for such protection on VBUS and USB data lines."
![screen shot 2017-01-27 at 15 46 26](https://cloud.githubusercontent.com/assets/281642/22363805/eb524116-e4a7-11e6-962e-c4ad2dde83b1.png)
- [x] "It is recommended to implement a resistor bridge to lower the voltage used for VBUS detection. This recommendation is even more important when VDD on the STM32 MCU is not present while VBUS is supplied from the connector." Done.
![screen shot 2017-01-27 at 15 44 52](https://cloud.githubusercontent.com/assets/281642/22363753/948987d6-e4a7-11e6-89ae-b0063c9688b6.png)
